### DISCUSSION TITLE
Minimal cookie management requirements when using YouTube/Twitter embeds

### DISCUSSION THREAD

**Discussion by setsqr** [orestbida/cookieconsent#376] [Q&A] [answered]
Hi,

I have a website on which I don't intentionally set/require any cookies. I don't use any Analytics or do anything (such as membership log-in) which requires any form of persistance. However the website does contain embedded Twitter timelines and YouTube videos which bring with them third-party cookies (I'm aware of the no-cookies YouTube embed URL but can't use it in this instance unfortunately).

The Twitter embed code is like this:

```
<a class="twitter-timeline" href="https://twitter.com/example">Tweets by example</a>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
```

This causes 2 cookies to be created: 'https://platform.twitter.com' and 'https://syndication.twitter.com'.

The YouTube embed code looks like this:

`<iframe width="560" height="315" src="https://www.youtube.com/embed/C0DPdy98e4c" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>`

This creates a cookie: 'https://www.youtube.com'.

I would like to maintain these embeds but need to do so in a legal (in the UK), GDPR-compliant way with minimal impact on the user experience.

What would be the minimal possible set up to do this? Do I definitely need a cookie manager to pop up in the UI or is there a way to just silently stop these services from persisting cookies by default without any UI interaction?

Many thanks for your help :)

**Comment by orestbida** [comment, marked as answer]
Hi @setsqr,

you can use [iframemanager](https://orestbida.com/demo-projects/iframemanager/demo1/) to embed tracking iframes while staying GDPR compliant. If you don't want to embed a consent tool, then you'd have to design a section/popup on your website, to allow users to opt-out in any moment.

Also, for the sake of clarity, GDPR does not apply to cookies only but to all types of tracking technologies. The fact that a websites has no cookie doesn't necessarily mean that it's GDPR compliant.

**Reply by setsqr** [reply]
Thanks @orestbida. My initial hope was that it might be possible to use these services whilst rejecting their cookies, but I realise that this is not the case: it's either embed the service and accept that it brings cookies with it, or don't embed the service. So I don't think there's any way around it other than to use an "intrusive" (meaning that in the best possible sense, as I appreciate the effort that has gone into making the cookieconsent modal attractive/accessible/easy etc) UI mechanism for explicit opt in. Thanks for the heads up on the iframemanager — we will almost certainly need to implement that too.

**Reply by orestbida** [reply]
> it's either embed the service and accept that it brings cookies with it, or don't embed the service

Sadly, that's the way it is.

**Comment by iamfredrik** [comment]
I would also like to be able to use Cookieconsent instead of iframemanager for handling iframe cookies. Please check the solutions offered by CookieBot and CookieFirst below if a similar feature could be added to this script:

https://support.cookiebot.com/hc/en-us/articles/360003790854-Iframe-cookie-consent-with-YouTube-example
https://support.cookiefirst.com/hc/en-us/articles/360011566458-Iframe-i-e-Video-players-only-load-when-consent-is-given


Edit:

Nevermind. The API can be used for this functionality. See https://github.com/orestbida/cookieconsent/issues/383#issuecomment-1264319277 how to use the API

Example:
```
const { cc } = useCookieConsent()
const [isConsent, setConsent] = React.useState(false)

const handleClick = (event) => {
    setConsent(true)
}

 React.useEffect(() => {
    if(cc.allowedCategory('marketing')) {
      setConsent(true)
    }
    if(!cc.allowedCategory('marketing') && isConsent) {
      cc.accept('marketing')
    }
  }, [isConsent])

return (
 <>
    {!isConsent && <div>
          <div dangerouslySetInnerHTML={{ __html: <p>This video content is provided by a third party. Accept marketing cookies to watch this video</p> }}/>
          <button
            className="consent-btn"
            type="button"
            onClick={handleClick}>
            Accept cookies
          </button>
     </div>}
    <iframe
          width="560"
          height="315"
          src={isConsent ? `https://www.youtube.com/embed/${video_id}?rel=0&modestbranding=1&autohide=1&mute=1&showinfo=0&controls=0&autoplay=1` : ''}
          title="YouTube video player"
          frameBorder="0"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen
      />
 </>
)
```
